front/cypress/e2e/content_builder/content_builder_preview_toggle.cy.ts

```diff
@@ -78,4 +78,22 @@ describe('Content builder mobile preview', () => {
       .contains('Edited text.Another edited text.')
       .should('be.visible');
   });
+
+  it('allows user to navigate between mobile and desktop preview and see content in both', () => {
+    cy.visit(`/admin/content-builder/projects/${projectId}/description`);
+    cy.get('#e2e-draggable-text').dragAndDrop('#e2e-content-builder-frame', {
+      position: 'inside',
+    });
+    cy.get('#e2e-text-box').click();
+    cy.get('#quill-editor').click();
+    cy.get('#quill-editor').clear().type('Sample text.', { force: true });
+
+    cy.get('#e2e-preview-toggle').find('input').click({ force: true });
+
+    getIframeBody().contains('Sample text.').should('be.visible');
+
+    cy.get('#e2e-desktop-preview').click();
+
+    getIframeBody().contains('Sample text.').should('be.visible');
```

Comment:
Not sure how you are checking that you are showing the desktop vs the mobile view here? What would be the easiest way to do this - maybe check on the iframe width?

Reply:
That is a very good question. I thought about it but I was heavily conflicted about it. It doesn't seem like an end to end test kind thing. It ties a lot into implementation detail. Ideally changing the width shouldn't be something that causes a test to fail 🤔

Reply:
This one of the odd ones. Normally it is good to test for user focused behaviour even with unit tests using react testing library. I think it might be better to leave this as is. It is testing that the user can access both flows (Desktop and Mobile). Going into the actual implementation will cause a brittle test whose failure might not exactly mean that the feature is broken

Reply:
I disagree it should remain as is - passing doesn't tell us that the feature is working, at the moment. I would suggest at least adding a data-cy attribute that indicates mobile vs desktop view and checking that it's correct in the test.

Reply:
Are you referring to avoiding css selectors in favour of (`data-cy`) to guarantee this?

Reply:
You can also do it with a classname or id, if you prefer. Just `data-cy` is standard for cypress tests, as far as I know. Also recommended in best practices: https://docs.cypress.io/guides/references/best-practices

Reply:
We use the same iFrame for both scenarios. The changes there are styling based

Reply:
Yes, I know. What you can do is add data-cy={mobilePreview? "mobile-preview-iframe:"desktop-preview-iframe"} to the iframe and then check that the correct data-cy attribute appears in the test. Because the styles rely on the same logic - you are at least indirectly testing that they are correct and that clicking between the two modes changes the view.